src/tls.c

```diff
@@ -786,6 +787,43 @@ static int connTLSWrite(connection *conn_, const void *data, size_t data_len) {
     return ret;
 }
 
+static int connTLSWritev(connection *conn_, const struct iovec *iov, int iovcnt) {
+    if (iovcnt == 1) return connTLSWrite(conn_, iov[0].iov_base, iov[0].iov_len);
+
+    /* Accumulate the amount of bytes of each buffer and check if it exceeds NET_MAX_WRITES_PER_EVENT. */
+    size_t cum = 0;
+    for (int i = 0; i < iovcnt; i++) {
+        cum += iov[i].iov_len;
+        if (cum > NET_MAX_WRITES_PER_EVENT) break;
+    }
+
+    /* The amount of all buffers is greater than NET_MAX_WRITES_PER_EVENT, 
+     * which is not worth doing so much memory copying to reduce system calls,
+     * therefore, invoke connTLSWrite() multiple times to avoid memory copies. */
+    if (cum > NET_MAX_WRITES_PER_EVENT) {
+        cum = 0;
```

Comment:
```suggestion
        int cum = 0;
```

Reply:
Please note that from hereafter the variable play a different role. To make it more readable, better to allocate new one (optimized by compiler).

Reply:
Done.